packages/react/src/components/variants-provider.component.tsx

```diff
@@ -77,9 +83,10 @@ const variantsScript = (variantsString: string, contentId: string) => `
       newParent.appendChild(winningTemplate.content.firstChild)
       parentNode.parentNode.replaceChild(newParent, parentNode);
     }
+  } else if (variants.length > 0) {
+    removeVariants()
   }
-})()
-`
+})()`.replace(/\s+/, ' ')
```

Comment:
mistake in my last comment - we'll want this to have `/g` so it matches all whitespace

```
})()`.replace(/\s+/g, ' ')
```